proxied-token/proxied-token-spec.ini

```diff
@@ -0,0 +1,295 @@
+
+[balanceOf-success]
+statusCode: _ => EVMC_SUCCESS
+output: _ => #asByteStackInWidth(BAL, 32)
++storage:
+    #hashedLocation({COMPILER}, {_BALANCES}, OWNER) |-> BAL
++requires:
+    andBool #rangeUInt(256, BAL)
+
+[balanceOf-failure]
+callValue: CALL_VALUE
++requires:
+    andBool #rangeUInt(256, CALL_VALUE)
+    andBool 0  <Int  CALL_VALUE
+
+[allowance]
+callData: #abiCallData("allowance", #address(OWNER), #address(SPENDER))
++requires:
+    andBool #rangeAddress(OWNER)
+    andBool #rangeAddress(SPENDER)
+
+[allowance-success]
+statusCode: _ => EVMC_SUCCESS
+output: _ => #asByteStackInWidth(ALLOWANCE, 32)
++storage:
+    #hashedLocation({COMPILER}, {_ALLOWANCES}, OWNER SPENDER) |-> ALLOWANCE
++requires:
+    andBool #rangeUInt(256, ALLOWANCE)
+
+[allowance-failure]
+callValue: CALL_VALUE
++requires:
+    andBool #rangeUInt(256, CALL_VALUE)
+    andBool 0  <Int  CALL_VALUE
+
+[approve]
+callData: #abiCallData("approve", #address(SPENDER), #uint256(VALUE))
++requires:
+    andBool #rangeAddress(SPENDER)
+    andBool #rangeUInt(256, VALUE)
+
+[approve-success]
+statusCode: _ => EVMC_SUCCESS
+output: _ => #asByteStackInWidth(1, 32)
+log: _:List ( .List => ListItem(#abiEventLog(PROXY_ID, "Approval", #indexed(#address(RANDOM_ID)), #indexed(#address(SPENDER)), #uint256(VALUE))) )
++storage:
+    #hashedLocation({COMPILER}, {_ALLOWANCES}, RANDOM_ID SPENDER) |-> (_:Int => VALUE)
+
+[approve-failure]
+callValue: CALL_VALUE
++requires:
+    andBool #rangeUInt(256, CALL_VALUE)
+    andBool 0  <Int  CALL_VALUE
+
+[transfer]
+statusCode: _ => EVMC_SUCCESS
+callData: #abiCallData("transfer", #address(TO_ID), #uint256(VALUE))
++requires:
+    andBool #rangeAddress(TO_ID)
+    andBool #rangeUInt(256, VALUE)
+    andBool #rangeUInt(256, BAL_FROM)
+    andBool #rangeUInt(256, BAL_TO)
+
+[transfer-success]
+output: _ => #asByteStackInWidth(1, 32)
+log: _:List ( .List => ListItem(#abiEventLog(PROXY_ID, "Transfer", #indexed(#address(RANDOM_ID)), #indexed(#address(TO_ID)), #uint256(VALUE))) )
+
+[transfer-success-1]
++storage:
+    #hashedLocation({COMPILER}, {_BALANCES}, RANDOM_ID) |-> (BAL_FROM => BAL_FROM -Int VALUE)
+    #hashedLocation({COMPILER}, {_BALANCES}, TO_ID)     |-> (BAL_TO   => BAL_TO   +Int VALUE)
++requires:
+    andBool RANDOM_ID =/=Int TO_ID
+    andBool VALUE <=Int BAL_FROM
+    andBool BAL_TO +Int VALUE <Int (2 ^Int 256)
+
+[transfer-success-2]
++storage:
+    #hashedLocation({COMPILER}, {_BALANCES}, RANDOM_ID) |-> BAL_FROM
++requires:
+    andBool RANDOM_ID ==Int TO_ID
+    andBool VALUE <=Int BAL_FROM
+    andBool BAL_FROM +Int VALUE <Int (2 ^Int 256)
+
+[transfer-failure]
+output: _ => #asByteStackInWidth(0, 32)
+[transfer-failure-1]
++storage:
+    #hashedLocation({COMPILER}, {_BALANCES}, RANDOM_ID) |-> (BAL_FROM => _)
+    #hashedLocation({COMPILER}, {_BALANCES}, TO_ID)     |->  BAL_TO
++requires:
+    andBool ( VALUE >Int BAL_FROM
+     orBool   BAL_TO +Int VALUE >=Int (2 ^Int 256) )
+
+[transfer-failure-2]
```

Comment:
how is this a failure?

Reply:
both transfer-failure-1 and transfer-failure-2 are failures because the math overflows, so the arithmetic checks would return false (see first few lines of transfer fn in deployed.sol)